datalake/src/test/java/com/sequenceiq/datalake/service/sdx/attach/SdxDetachServiceTest.java

```diff
@@ -0,0 +1,58 @@
+package com.sequenceiq.datalake.service.sdx.attach;
+
+import com.sequenceiq.cloudbreak.auth.crn.RegionAwareCrnGenerator;
+import com.sequenceiq.datalake.entity.SdxCluster;
+import com.sequenceiq.datalake.repository.SdxClusterRepository;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import static org.mockito.ArgumentMatchers.any;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import static org.mockito.Mockito.when;
+import org.mockito.MockitoAnnotations;
+
+public class SdxDetachServiceTest {
```

Comment:
please write unit test for detachStack and detachExternalDatabase too

Reply:
So for these, the commands detachStack and detachExternalDatabase basically just run one command:

- stackV4Endpoint.updateNameAndCrn for detachStack
- redbeamsServerEndpoint.updateClusterCrn for detachExternalDatabase

As far as I know, unit tests for these would just mock the endpoints and have them return the correct value, so I don't really see much value in having these tests. Please correct me if I'm wrong though.

Reply:
yes, right, now it would bring not much value to creating these unit tests, you can skip this if you want, although writing a simple unit test would also prevent anyone else to break these methods easily in the future and it also would increase test coverage

Reply:
Thanks for the thorough explanation. I can now see the value in adding these just to prevent future issues, I'll add these in.

Reply:
I went ahead and added a lot more tests for both the detachService and the attachService. Let me know if they look good.